plugins/modem-manager/fu-mm-device.c

```diff
@@ -275,6 +315,18 @@ fu_mm_device_probe_default (FuDevice *device, GError **error)
 		return FALSE;
 	}
 
+	/* Flag for mhi-based mbim device */
```

Comment:
I don't think the logic behind this flag is ok. Do you only need the is_mhi_dev flag to build a vendor ID with PCI prefix instead of USB prefix?

I think that instead of this logic we could use this patch I wrote last December, which we didn't integrate yet because we were waiting for the actual final setup to be ready, but it would probably be worth getting merged to make it work also with this new QDU method: https://github.com/aleksander0m/fwupd/commit/abf7685e8e233be01da4f677508f88a690ff7ecf

Could you see if that patch helps you to do the same?

Reply:
Also, this one may be worth including: https://github.com/aleksander0m/fwupd/commit/5e936040f21bc3084ddb4402d14457c1ff139b4c

Reply:
> but it would probably be worth getting merged to make it work also with this new QDU method: [aleksander0m@abf7685](https://github.com/aleksander0m/fwupd/commit/abf7685e8e233be01da4f677508f88a690ff7ecf)
> 
> Could you see if that patch helps you to do the same?

It looks great in your patch, could you tell me when it can be merged in?

Reply:
I know there is at least one additional team working on top of that patch, so maybe it's worth merging that already so that everyone can work on top of it? @hughsie what do you think? should I suggest that single patch (adding generic PCI support for the MM plugin) in a separate pull request?

Reply:
I have created this pull request with that commit after rebasing them on top of git master: https://github.com/fwupd/fwupd/pull/3210